app/models/Classroom.coco.coffee

```diff
@@ -273,3 +274,44 @@ module.exports = class Classroom extends CocoModel
       return classroomUtils.getDisplayPermission('write')
     else if @hasReadPermission()
       return classroomUtils.getDisplayPermission('read')
+
+  revokeStudentLicenses: ->
+    students = new Users()
+    Promise.all(students.fetchForClassroom(@, {removeDeleted: true, data: {project: 'firstName,lastName,name,email,products,deleted'}}))
+      .then =>
+        studentsToRevoke = students.models.filter((student) => student.prepaidStatus() is 'enrolled' and student.prepaidType() is 'course')
+        if studentsToRevoke.length > 0
+          @showRevokeConfirm(studentsToRevoke).then (revokeConfirmed) =>
+            return unless revokeConfirmed
+            students.models.forEach (student) =>
+              student.revokePrepaid @, () =>
+                noty
+                  text: $.i18n.t('teacher.license_revoked_from_student') + student.broadName()
+                  type: 'information',
+                  timeout: 2000,
+                  layout:'topCenter'
+
+  showRevokeConfirm: (studentsToRevoke)->
+    new Promise((resolve) ->
+      notification = noty
+        text: studentsToRevoke.length + $.i18n.t 'teacher.archive_revoke_confirm'
+        type: 'info',
+        layout: 'center',
+        buttons: [
+          {
+            addClass: 'btn btn-primary'
+            text: $.i18n.t 'modal.okay'
```

Comment:
This should say "Revoke and Archive".

I think we should switch the button order, so the non-revoke action is on the left and the revoke action is on the right.